test/ephemeral-keys.spec.js

```diff
@@ -1,44 +1,56 @@
+/* eslint max-nested-callbacks: ["error", 8] */
 /* eslint-env mocha */
 'use strict'
 
 const expect = require('chai').expect
-const EC = require('elliptic').ec
+const parallel = require('async/parallel')
 
-const crypto = require('../src')
 const fixtures = require('./fixtures/go-elliptic-key')
+const crypto = require('../src')
 
-describe('generateEphemeralKeyPair', () => {
-  it('returns a function that generates a shared secret', () => {
-    const res = crypto.generateEphemeralKeyPair('P-256')
-    const ourPublic = '044374add0df35706db7dade25f3959fc051d2ef5166f8a6a0aa632d0ab41cdb4d30e1a064e121ac56155235a6b8d4c5d8fe35e019f507f4e2ff1445e229d7af43'
-
-    expect(
-      res.genSharedKey(ourPublic)
-    ).to.have.length(32)
+const curves = ['P-256', 'P-384', 'P-521']
+const lengths = {
+  'P-256': 65,
+  'P-384': 97,
+  'P-521': 133
```

Comment:
these lengths are not the same as : https://github.com/libp2p/js-libp2p-crypto/pull/12/files#r83008550